mobile/src/main/java/org/openhab/habdroid/background/AlarmChangedReceiver.java

```diff
@@ -0,0 +1,93 @@
+package org.openhab.habdroid.background;
+
+import android.app.AlarmManager;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.Log;
+import androidx.work.Constraints;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import org.openhab.habdroid.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.background.BackgroundUtils.WORKER_TAG_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ENABLED;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ITEM;
+
+public class AlarmChangedReceiver extends BroadcastReceiver {
+    private static final String TAG = AlarmChangedReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d(TAG, "onReceive() with intent " + intent.getAction());
+
+        if (AlarmManager.ACTION_NEXT_ALARM_CLOCK_CHANGED.equals(intent.getAction())
+                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+            if (prefs.getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, false)) {
+                startAlarmChangedWorker(context);
+            }
+        }
+    }
+
+    public static void startAlarmChangedWorker(Context context) {
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+        startAlarmChangedWorker(context, prefs.getString(PREFERENCE_ALARM_CLOCK_ITEM, ""));
+    }
+
+    public static void startAlarmChangedWorker(Context context, String itemName) {
```

Comment:
Maybe pass in the enabled flag here as well and rename the function appropriately?
```
public static void startAlarmChangedWorkerIfNeeded(Context context, boolean enabled, String itemName) {
    if (!enabled || TextUtils.isEmpty(itemName.trim()) {
        return;
    }
    [...]
}

That way you don't need to check the enabled flag in all calling places.

Reply:
Done